testutil/beaconmock/beaconmock_test.go

```diff
@@ -64,3 +71,77 @@ func TestAttestationData(t *testing.T) {
 	require.NoError(t, err)
 	testutil.RequireGoldenJSON(t, attData)
 }
+
+func TestBeaconCommitteeSubscriptions(t *testing.T) {
+	ctx := context.Background()
+
+	const (
+		slotA = 123
+		slotB = 456
+		vIdxA = 1
+		vIdxB = 2
+		vIdxC = 3
+	)
+
+	aggregators := map[eth2p0.Slot]eth2p0.ValidatorIndex{
+		slotA: vIdxA,
+		slotB: vIdxB,
+	}
+
+	bmock, err := beaconmock.New(beaconmock.WithAttestationAggregation(aggregators))
+	require.NoError(t, err)
+
+	expected := []*eth2exp.BeaconCommitteeSubscriptionResponse{
+		{ValidatorIndex: vIdxA, IsAggregator: true},
+		{ValidatorIndex: vIdxB, IsAggregator: true},
+		{ValidatorIndex: vIdxC, IsAggregator: false},
+	}
+
+	subs := []*eth2exp.BeaconCommitteeSubscription{
+		{
+			Slot:             slotA,
+			ValidatorIndex:   vIdxA,
+			CommitteesAtSlot: rand.Uint64(),
+			CommitteeIndex:   eth2p0.CommitteeIndex(rand.Uint64()),
+			SlotSignature:    testutil.RandomEth2Signature(),
+		},
+		{
+			Slot:             slotB,
+			ValidatorIndex:   vIdxB,
+			CommitteesAtSlot: rand.Uint64(),
+			CommitteeIndex:   eth2p0.CommitteeIndex(rand.Uint64()),
+			SlotSignature:    testutil.RandomEth2Signature(),
+		},
+		{
+			Slot:             slotA,
+			ValidatorIndex:   vIdxC,
+			CommitteesAtSlot: rand.Uint64(),
+			CommitteeIndex:   eth2p0.CommitteeIndex(rand.Uint64()),
+			SlotSignature:    testutil.RandomEth2Signature(),
+		},
+	}
+
+	vapi, err := validatorapi.NewComponentInsecure(t, bmock, 0)
```

Comment:
not sure this test should be here? Why not `validatorapi`?

Reply:
well this test is for testing beaconmock flow through validatorapi and httpwrap, also validatorapi doesn't have implementation for this (yet) so not sure.

Reply:
can rename this test to `TestBeaconCommitteeSubscriptionsV2Integration` maybe

Reply:
I would add this to `router_internal_test` to test that flow.